contracts/Marketplace.sol

```diff
@@ -0,0 +1,164 @@
+
+    struct Auction {
+        uint256 tokenId;
+        address seller;
+        uint256 startTime;
+        uint256 endTime;
+        uint256 reservePrice;
+        bool started;
+    }
+
+    mapping(uint256 => Auction) public auctions;
+    uint256 public nextAuctionId = 0;
+
+    struct HighestBidder {
+        address bidder;
+        uint256 bid;
+        uint256 bidTime;
+    }
+    mapping(uint256 => HighestBidder) public highestBids;
+
+
+    constructor(address _assetNft) {
+        assetNftContract = Fisch(_assetNft);
+    }
+
+    function fetchAuction() public {}
+
+    function buyFixedSale() public payable {}
+
+    function cancelFixedSale() public {}
+
+    function startAuction(
+        uint256 tokenId,
+        uint256 startTime,
+        uint256 endTime,
+        uint256 reservePrice
+    ) external nonReentrant{
+        uint256 auctionId = nextAuctionId++;
+
+        auctions[auctionId] = Auction(
+            tokenId,
+            msg.sender,
+            startTime,
+            endTime,
+            reservePrice,
+            true
+        );
+
+        emit AuctionCreated(
+            auctionId,
+            tokenId,
+            msg.sender,
+            startTime,
+            endTime,
+            reservePrice,
+            true
+        );
+    }
+
+    function cancelAuction(uint256 _auctionId) public {
+        require(auctions[_auctionId].seller == msg.sender, "You are not the seller");
+
+        auctions[_auctionId].started = false;
+        emit AuctionCancelled(
+            _auctionId,
+            auctions[_auctionId].tokenId,
+            msg.sender
+        );
+    }
+
+    // Function to receive Ether. msg.data must be empty
+    receive() external payable {
+        emit AmountReceived(msg.sender, msg.value);
+    }
+
+    // Fallback function is called when msg.data is not empty
+    fallback() external payable {}
+
+    function sendViaCall(address payable _to, uint256 _amount) public payable {
+        // Call returns a boolean value indicating success or failure.
+        // This is the current recommended method to use.
+        (bool sent, /*bytes memory data*/) = _to.call{value: _amount}("");
+        require(sent, "Failed to send Matic");
+        emit AmountSent(_to, _amount);
+    }        
+
+    function placeBid(uint256 _auctionId) public payable nonReentrant {
+
+        require(block.timestamp < auctions[_auctionId].endTime, "Auction has ended");
+        require(msg.sender != auctions[_auctionId].seller, "you should are not allowed to place a bid on your own auction.");
+        require(msg.sender != address(0), "this is the address 0x00.. address");
+
+        HighestBidder memory highestBidder = highestBids[_auctionId];
+        uint256 minBid = highestBidder.bid.add(minimumBid);
+        if (highestBidder.bid == 0) {
+            minBid = highestBids[_auctionId].bid.add(minimumBid);
+        }
+
+        require(msg.value > minBid, "Has not exceeded the best bid");
```

Comment:
I think here it should be greater than or equal to ">=", so that the user can place the exact bid and still pass.

Reply:
I'll make this change